projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.pipes.ts

```diff
@@ -322,12 +319,12 @@ export class IgxTreeGridNormalizeRecordsPipe implements PipeTransform {
         this.gridAPI = gridAPI as IgxTreeGridAPIService;
     }
 
-    transform(collection: any[], pipeTrigger: number): any[] {
+    public transform(collection: any[], _: number): any[] {
         const grid =  this.gridAPI.grid;
         const primaryKey = grid.primaryKey;
         // using flattened data because origin data may be hierarchical.
-        const flatData = grid.flatData;
-        const res = flatData.map(rec =>
+        collection = grid.flatData;
+        const res = collection.map(rec =>
```

Comment:
I am not sure if setting the `collection` parameter won't break anything so it is better to change it to underscore.